File: 11_ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right , maxArea = 0, len(height)-1, 0
        while left < right:
            if height[left] < height[right]:
                maxArea = max(maxArea, height[left]*(right-left))
                left += 1
            else:
                maxArea = max(maxArea, height[right]*(right-left))
                right -= 1
        return maxArea

File: C#/11_ContainerWithMostWater.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left = 0, right = height.Length - 1, maxArea = 0;
        while(left < right){
            if (height[left] < height[right]){
                maxArea = Math.Max(maxArea, height[left]*(right-left));
                left++;
            } else{
                maxArea = Math.Max(maxArea, height[right]*(right-left));
                right--;
            }
        }
        return maxArea;
    }
}